indico/modules/events/timetable/forms.py

```diff
@@ -83,3 +84,41 @@ def __init__(self, *args, **kwargs):
         self.day = kwargs.pop('day')
         kwargs['start_date'] = find_earliest_gap(kwargs['event'], day=self.day, duration=duration)
         super(ContributionEntryForm, self).__init__(*args, **kwargs)
+
+
+class SessionBlockEntryForm(SessionBlockForm):
+    time = TimeField(_("Time"), description=_("Time where the session block will be scheduled."))
+    duration = TimeDeltaField(_("Duration"), [DataRequired(), MaxDuration(timedelta(hours=24))],
+                              default=timedelta(minutes=60), units=('minutes', 'hours'),
+                              description=_("The duration of the session block"))
+
+    def __init__(self, *args, **kwargs):
+        self.event = kwargs.pop('event')
+        self.day = kwargs.pop('day')
+        duration = self.duration.kwargs['default']
+        start_dt = find_earliest_gap(self.event, self.day, duration=duration)
+        kwargs['time'] = start_dt.astimezone(self.event.tzinfo).time() if start_dt else None
+        super(SessionBlockEntryForm, self).__init__(*args, **kwargs)
+
+    @property
+    def data(self):
+        data = super(SessionBlockEntryForm, self).data
+        del data['time']
+        return data
+
+    @generated_data
+    def start_dt(self):
+        dt = datetime.combine(self.day, self.time.data)
+        return self.event.tzinfo.localize(dt).astimezone(utc)
+
+    def validate_duration(self, field):
+        end_dt = self.start_dt.data + field.data
+        if end_dt > self.event.end_dt:
+            raise ValidationError(_("With current time and duration the session block ends after the event."))
```

Comment:
the messages don't sound very well. "Session block exceeds the event end time" or something like that?